docs/integration-guide.md

```diff
@@ -107,6 +107,239 @@ Here's how we recommend doing it:
+        expected_subject = u"Hello World";
+        expected_message = u"Hello! I am happy to be here! :smile: \nThe Wikipedia featured article for today is **[Marilyn Monroe](https://en.wikipedia.org/wiki/Marilyn_Monroe)**";
+        self.send_and_test_stream_message('hello', expected_subject, expected_message,content_type="application/x-www-form-urlencoded")
+
+    def get_body(self, fixture_name):
+        return self.fixture_data("helloworld", fixture_name, file_type="json")
+
+```
+
+And to simulate the post data of a real request, we add the fixture `zerver/fixtures/helloworld/hello_world.json` with this data:
+
+```
+{
+  "type":"wikipedia",
+  "day":"Wed, 6/1",
+  "featured_title":"Marilyn Monroe",
+  "featured_url":"https://en.wikipedia.org/wiki/Marilyn_Monroe",
+}
+```
+
+Now you can run these tests from within the Zulip dev environment with this
+command:
+
+```
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:/srv/zulip$
+./tools/test-backend zerver.tests.test_hooks.HelloWorldHookTests
+```
+
+(Note: You must run the tests from `/srv/zulip` directory.)
+
+You will see some script output and if all the tests have passed, you will see:
+
+```
+Running zerver.tests.test_hooks.HelloWorldHookTests.test_hello_message
+DONE!
+```
+
+#### Step 4: Create documentation
+
+Next, we add end-user documentation for our webhook integration to
+`templates/zerver/integrations.html`.
+
+First, add a `div` that displays the logo of your integration and a link to its
+documentation:
+
+```
+ <div class="integration-lozenge integration-helloworld">
+   <a class="integration-link integration-helloworld" href="#helloworld">
+      <img class="integration-logo" src="/static/images/integrations/logos/helloworld.png" alt="Hello World logo" />
+      <span class="integration-label">Hello World</span>
+   </a>
+ </div>
+```
+
+And second, a div with the usage instructions:
+
+```
+<div id="helloworld" class="integration-instructions">
+
+    <p>Learn how Zulip integrations work with this simple Hello World example!</p>
+
+    <p>The Hello World webhook will use the <code>test<code> stream, which is
+    created by default in the Zulip dev environment. If you are running
+    Zulip in production, you should make sure this stream exists.</p>
+
+    <p>Next, on your <a href="/#settings" target="_blank">Zulip
+    settings page</a>, create a Hello World bot.  Construct the URL for
+    the Hello World bot using the API key and stream name:
+      <code>{{ external_api_uri }}/v1/external/helloworld?api_key=abcdefgh&amp;stream=test</code>
+    </p>
+
+    <p><b>Congratulations! You're done!</b><br /> Your messages may look like:</p>
+
+    <img class="screenshot" src="/static/images/integrations/helloworld/001.png" />
+</div>
+```
+
+Both blocks should fall alphabetically so we add these two divs between the
+blocks for Github and Hubot, respectively.
+
+See [Documenting your integration](#documenting-your-integration) for further
+details.
```

Comment:
maybe add "further details, especially on how to easily create the message screenshot"